src/frontend/src/pages/TeamsPage/DescriptionPageBlock.tsx

```diff
@@ -0,0 +1,125 @@
+import { isUnderWordCount, countWords } from 'shared/src/word-count';
+
+interface DescriptionPageBlockProps {
+  team: Team;
+}
+
+const DescriptionPageBlock: React.FC<DescriptionPageBlockProps> = ({ team }) => {
+  const auth = useAuth();
+  const theme = useTheme();
+  const [isEditingDescription, setIsEditingDescription] = useState(false);
+  const [description, setDescription] = useState(team.description);
+  const [isPreview, setIsPreview] = useState(false);
+  const { isLoading, isError, error, mutateAsync } = useEditTeamDescription(team.teamId);
+
+  if (isError) return <ErrorPage message={error?.message} />;
+  if (isLoading) return <LoadingIndicator />;
+
+  const handleSubmit = async () => {
+    if (!isUnderWordCount(description, 300)) {
+      return alert('Description must be less than 300 words');
+    }
+    await mutateAsync(description);
+    resetDefaults();
+  };
+
+  const resetDefaults = () => {
+    setIsEditingDescription(false);
+    setIsPreview(false);
+  };
+
+  const hasPerms = () => {
+    return !(
+      auth.user &&
+      (auth.user.role === 'ADMIN' || auth.user.role === 'APP_ADMIN' || auth.user.userId === team.leader.userId)
+    );
+  };
+
+  const headerRight = (
+    <div style={{ display: 'flex' }}>
+      <Button
+        onClick={() => {
+          resetDefaults();
+        }}
+      >
+        Cancel
+      </Button>
+      <Button
+        onClick={() => setIsPreview(!isPreview)}
+        sx={{
+          ml: 2,
+          backgroundColor: theme.palette.grey[600],
+          color: theme.palette.getContrastText(theme.palette.grey[600]),
+          '&:hover': {
+            backgroundColor: theme.palette.grey[700]
+          }
+        }}
+      >
+        {isPreview ? 'Edit' : 'Preview'}
+      </Button>
+      <Button
+        sx={{
+          ml: 2,
+          backgroundColor: theme.palette.success.main,
+          color: theme.palette.success.contrastText,
+          '&:hover': {
+            backgroundColor: theme.palette.success.dark
+          }
+        }}
+        onClick={handleSubmit}
+      >
+        Save
+      </Button>
+    </div>
+  );
+
+  const editingView = (
+    <PageBlock title={'Description'} headerRight={headerRight}>
+      {isPreview ? (
+        <ReactMarkdown className={styles.markdown}>{description}</ReactMarkdown>
+      ) : (
+        <TextField
+          fullWidth
+          multiline
+          rows={10}
+          value={description}
+          onChange={(e) => setDescription(e.target.value)}
+          inputProps={{
+            maxlength: isUnderWordCount(description, 300) ? null : 0
+          }}
+          helperText={`${countWords(description)}/300 words`}
+          error={!isUnderWordCount(description, 300)}
+        />
+      )}
+    </PageBlock>
+  );
+
+  const nonEditingView = (
+    <PageBlock
+      title={'Description'}
+      headerRight={<IconButton disabled={hasPerms()} onClick={() => setIsEditingDescription(true)} children={<Edit />} />}
```

Comment:
Can you make it so the edit button just doesn't even show up if you don't have perms, instead of just disabling it, I think it'll look a bit cleaner